radicle-httpd/src/api/v1/projects.rs

```diff
@@ -224,16 +227,15 @@ async fn activity_handler(
 /// `GET /projects/:project/tree/:sha/*path`
 async fn tree_handler(
     Extension(ctx): Extension<Context>,
-    Path((project, sha, path)): Path<(Id, Oid, String)>,
+    Path((_, sha, path)): Path<(Id, Oid, String)>,
 ) -> impl IntoResponse {
     let path = path.strip_prefix('/').ok_or(Error::NotFound)?.to_string();
     let storage = &ctx.profile.storage;
-    let repo = storage.repository(project)?;
-    let tree = radicle_surf::object::tree(&repo.raw().into(), Some(Sha { sha }), Some(path))?;
+    let repo = Repository::open(storage.path())?;
+    let sub_dir = repo.directory(sha, &path)?;
     let response = json!({
-        "path": &tree.path,
-        "entries": &tree.entries,
-        "info": &tree.info,
```

Comment:
My first check was it seems not really needed. I will list the actual content, and if needed, we can add it.

Reply:
I removed the `Info` type and flattened it. The serialization for `Tree` is now [this](https://github.com/radicle-dev/radicle-git/blob/main/radicle-surf/src/source/object/tree.rs#L87-L101). So `Directory` isn't good enough in this case since it's missing information that would usually be displayed.